policy-controller/k8s/index/src/pod.rs

```diff
@@ -68,6 +68,68 @@ pub(crate) fn tcp_port_names(spec: Option<k8s::PodSpec>) -> HashMap<String, Port
     port_names
 }
 
+/// Gets the container probe ports for a Pod.
+///
+/// The result is a mapping for each probe port exposed by a container in the
+/// Pod and the paths for which probes are expected. When a TCP probe is
+/// configured there is no path, so we use '/'.
+pub(crate) fn get_container_probes(
+    spec: &Option<k8s::PodSpec>,
+    port_names: &HashMap<String, PortSet>,
+) -> PortMap<HashSet<String>> {
+    let mut probes = PortMap::<HashSet<String>>::default();
+    for container in spec.iter().flat_map(|s| s.containers.iter()) {
+        set_probe_ports(&container.liveness_probe, &mut probes, port_names);
+        set_probe_ports(&container.readiness_probe, &mut probes, port_names);
+        set_probe_ports(&container.startup_probe, &mut probes, port_names);
+    }
+    probes
+}
+
+fn set_probe_ports(
+    probe: &Option<k8s::Probe>,
+    probes: &mut PortMap<HashSet<String>>,
+    port_names: &HashMap<String, PortSet>,
+) {
+    let probe = match probe {
+        Some(probe) => probe,
+        None => return,
+    };
+    if let Some(http) = &probe.http_get {
+        if let Some(path) = &http.path {
+            match &http.port {
+                k8s::IntOrString::Int(port) => {
+                    if let Ok(p) = u16::try_from(*port).and_then(NonZeroU16::try_from) {
+                        let paths = probes.entry(p).or_default();
+                        paths.insert(path.clone());
+                    }
+                }
+                k8s::IntOrString::String(port) => {
+                    for p in port_names.get(port).into_iter().flatten() {
+                        let paths = probes.entry(*p).or_default();
+                        paths.insert(path.clone());
+                    }
+                }
+            }
+        }
+    } else if let Some(tcp) = &probe.tcp_socket {
+        match &tcp.port {
+            k8s::IntOrString::Int(port) => {
+                if let Ok(p) = u16::try_from(*port).and_then(NonZeroU16::try_from) {
+                    let paths = probes.entry(p).or_default();
+                    paths.insert("/".to_string());
```

Comment:
Rather than using "/" for TCP, it would be more natural to have this be an Option and use None.

Reply:
I think we probably want to ignore TCP probe ports completely. If we were to automatically authorize them, we're likely to expose application ports.